### PR TITLE
Added authentication service to check and store (localstorage) a token

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Added authentication service to check and store (localstorage) a token. On login, the core's authorize API is called which in turns carries out the relevant Git validations and authorization. Core then redirects to the referrer URL with a token. This token gets stored locally and gets  retrieved for subsequent page visits. On logout the token gets cleared.

---
This change is [Reviewable](https://reviewable.io/reviews/almighty/almighty-ui/76)

### CODE REVIEW

**Comment by aslakknutsen** [comment]
Hold merge this until https://github.com/almighty/almighty-core/pull/229 is deployed.

**Comment by sanbornsen** [comment]
Can we use private variable instead of local storage to store auth key? Local storage is globally accessible from the console.

**Comment by joshuawilson** [comment]
@aslakknutsen what is the status of the core PR 229?

**Comment by nimishamukherjee** [comment]
@sanbornsen Localstorage is used to persist the token. How do we use private variable in this scenario?

**Comment by sanbornsen** [comment]
@nimishamukherjee 

This should be the general flow - 
* Make a authentication service (You already made it)
* Define a private variable as **token**.
* Define two functions, **setToken**(private) and **getToken**(public)
* Add Authentication service into providers list at app.module.ts
* Initiate Authentication service at constructors for any other service where you want to use the token.

Let's discuss !

File: src/app/auth/authentication.service.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { Router } from "@angular/router";
+import { Logger } from '../shared/logger.service';
+
+@Injectable()
+export class AuthenticationService {
+  private authToken: any = null;
+
+  constructor(private router: Router, private logger: Logger) {
+    //If a token is passed part of the url - extract that and set it    
+    if((window.location.href).indexOf('?token=')>0){
```
Comment (bartoszmajsak) [inline src/app/auth/authentication.service.ts]: What if `token` won't come as first query param? Wouldn't it be better to extract all first into some sort of map and get relevant by key?

File: .gitignore [hunk 2]
```diff
@@ -26,6 +26,7 @@ npm-debug.log
 testem.log
 /typings
 functional_tests.log
+unit_tests.log
```
Comment (sanbornsen) [inline .gitignore]: how come this change is here?

File: README.adoc [hunk 3]
```diff
@@ -17,6 +17,21 @@ variable "ENV" to "inmemory" and do a rebuild. The setting is stored inside the
...
+
+----
+./run_unit_tests.sh
+----
+
+For unit testing, Jasmine, Karma, Webpack is used. There are unit tests available for 
+both services and UI compnents. There are `.spec.ts` files available in individual 
+component or service folder. Karma and Webpack config is written inside config directry 
+under root.
+
```
Comment (sanbornsen) [inline README.adoc]: how come this change is here?

File: cico_build_deploy.sh [hunk 4]
```diff
@@ -15,6 +15,16 @@ mkdir -p dist && docker run --detach=true --name=almighty-ui-builder -e "API_URL
...
+## Exec unit tests
+docker exec almighty-ui-builder ./run_unit_tests.sh
+
+if [ $? -eq 0 ]; then
+  echo 'CICO: unit tests OK'
+else
+  echo 'CICO: unit tests FAIL'
+  exit 1
+fi
+
```
Comment (sanbornsen) [inline cico_build_deploy.sh]: how come this change is here?

**Comment by joshuawilson** [comment]
[test]

**Comment by sanbornsen** [comment]
We should close this PR for now. I have created a parallel branch  **almighty-ui-gitlogin** with these changes and everything is up to date there. We should use that branch to work on login stuff and finally merge that with master once we are done and everything in place.

**Comment by joshuawilson** [comment]
Closing this PR since the branch it is based on has moved. @sanbornsen please create a new PR.

**Pull request closed without merging**